### PR TITLE
Add support for Qwen3 MoE models

### LINKED ISSUE #705: Support Qwen3-MoE
### 🚀 The feature, motivation and pitch

https://github.com/huggingface/transformers/tree/main/src/transformers/models/qwen3_moe

See how to handle aux loss for MoE:
https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/transformers/model/mixtral.py

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chiwanpark**
## Summary

This PR adds Qwen3 MoE model support, and fixes #705.

The MLP layer for Qwen3 MoE optionally receives `intermediate_size` and overrides it to that in config ([link](https://github.com/huggingface/transformers/blob/v4.51.3/src/transformers/models/qwen3_moe/modular_qwen3_moe.py#L58)), while `LigerSwiGLUMLP` does not receive the `intermediate_size`; thus, I've created a custom MLP module (`Qwen3MoeSwiGLUMLP`) like Phi 3.

## Testing Done

I've added tests for Qwen3 MoE, and confirmed that it works correctly. The tolerance thresholds are copied from that of Qwen3.

```
...
test/convergence/fp32/test_mini_models.py::test_mini_model[mini_qwen3-32-0.0001-dtype5-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                                        [ 35%] 
test/convergence/fp32/test_mini_models.py::test_mini_model[mini_qwen3_moe-32-0.0001-dtype6-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                                    [ 41%] 
test/convergence/fp32/test_mini_models.py::test_mini_model[mini_qwen2_vl-32-0.0001-dtype7-1e-05-0.1-0.005-1e-05-0.005-1e-05] PASSED                                       [ 47%] 
...
```

```
...
test/convergence/fp32/test_mini_models_with_logits.py::test_mini_model[mini_qwen3-32-0.0001-dtype5-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                            [ 35%]
test/convergence/fp32/test_mini_models_with_logits.py::test_mini_model[mini_qwen3_moe-32-0.0001-dtype6-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                        [ 41%]
test/convergence/fp32/test_mini_models_with_logits.py::test_mini_model[mini_qwen2_vl-32-0.0001-dtype7-1e-08-2e-05-0.005-1e-05-0.005-1e-05] PASSED                         [ 47%]
...
```

```
...
test/convergence/bf16/test_mini_models.py::test_mini_model[mini_qwen3-32-0.0001-dtype5-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                                              [ 37%]
test/convergence/bf16/test_mini_models.py::test_mini_model[mini_qwen3_moe-32-0.0001-dtype6-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                                          [ 43%]
test/convergence/bf16/test_mini_models.py::test_mini_model[mini_qwen2_vl-32-0.0001-dtype7-0.001-0.05-0.1-0.01-0.01-0.01] PASSED                                           [ 50%]
...
```

```
...
test/convergence/bf16/test_mini_models_with_logits.py::test_mini_model[mini_qwen3-32-0.0001-dtype5-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                                  [ 37%]
test/convergence/bf16/test_mini_models_with_logits.py::test_mini_model[mini_qwen3_moe-32-0.0001-dtype6-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                              [ 43%]
test/convergence/bf16/test_mini_models_with_logits.py::test_mini_model[mini_qwen2_vl-32-0.0001-dtype7-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                               [ 50%]
...
```

- Hardware Type: A100 80GB 4 cards
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: test/convergence/bf16/test_mini_models.py [hunk 1]
```diff
@@ -395,6 +399,41 @@
         ),
     )
 
+    MINI_MODEL_SETUPS["mini_qwen3_moe"] = MiniModelConfig(
+        liger_kernel_patch_func=apply_liger_kernel_to_qwen3_moe,
+        liger_kernel_patch_revert_func=revert_liger_kernel_to_qwen3_moe,
+        model_class=Qwen3MoeForCausalLM,
+        mini_model_config=Qwen3MoeConfig(
+            vocab_size=151936,
```
Comment (Tcc0403) [inline test/convergence/bf16/test_mini_models.py:407]:
```suggestion
            vocab_size=32000,  # 151936
```

**Comment by Tcc0403** [comment]
Thank you, lgtm! Lower vocab_size in convergence tests then we're good to merge.

**Review by Tcc0403 — APPROVED** [review]
